ragna/core/_rag.py

```diff
@@ -147,66 +211,33 @@ def __init__(
         self.params = params
         self._unpacked_params = self._unpack_chat_params(params)
 
-        self._prepared = False
         self._messages: list[Message] = []
 
-    async def prepare(self) -> Message:
-        """Prepare the chat.
-
-        This [`store`][ragna.core.SourceStorage.store]s the documents in the selected
-        source storage. Afterwards prompts can be [`answer`][ragna.core.Chat.answer]ed.
-
-        Returns:
-            Welcome message.
-
-        Raises:
-            ragna.core.RagnaException: If chat is already
-                [`prepare`][ragna.core.Chat.prepare]d.
-        """
-        if self._prepared:
-            raise RagnaException(
-                "Chat is already prepared",
-                chat=self,
-                http_status_code=400,
-                detail=RagnaException.EVENT,
-            )
-
-        await self._run(self.source_storage.store, self.documents)
-        self._prepared = True
-
-        welcome = Message(
-            content="How can I help you with the documents?",
-            role=MessageRole.SYSTEM,
-        )
-        self._messages.append(welcome)
-        return welcome
-
     async def answer(self, prompt: str) -> Message:
         """Answer a prompt.
 
         Returns:
             Answer.
-
-        Raises:
-            ragna.core.RagnaException: If chat is not
-                [`prepare`][ragna.core.Chat.prepare]d.
         """
-        if not self._prepared:
-            raise RagnaException(
-                "Chat is not prepared",
-                chat=self,
-                http_status_code=400,
-                detail=RagnaException.EVENT,
-            )
+
+        # TODO: Add welcome message if message list is empty
 
         prompt = Message(content=prompt, role=MessageRole.USER)
         self._messages.append(prompt)
 
-        sources = await self._run(
-            self.source_storage.retrieve, self.documents, prompt.content
+        kwargs = self._unpacked_params[self.corpus.source_storage.retrieve]
+        sources = await _run(
+            self.corpus.source_storage.retrieve,
+            self.corpus.documents,
```

Comment:
`SourceStorage.store` and `SourceStorage.retrieve` should take a `Corpus` rather than a list of documents.